tests/test.sitemap.js

```diff
@@ -123,6 +123,43 @@ tap.test('ignore routes by plugin config on route', async(t) => {
   t.end();
 });
 
+tap.test('accepts dynamic route options', async(t) => {
+  const server = await new Hapi.Server({ port: 8080 });
+  server.route({
+    method: 'get',
+    path: '/path/{param}',
+    handler(request, h) {
+      return 'hello';
+    }
+  });
+  await server.register({
+    plugin,
+    options: {
+      dynamicRoutes: {
+        '/path/{param}': async () => {
+          return [
+            '/path/param1',
+            '/path/param2',
+            '/path/param3'
+          ];
+        }
+      }
+    }
+  });
+  await server.start();
+  const response = await server.inject({
+    method: 'get',
+    url: '/sitemap.json'
+  });
+  t.equal(response.statusCode, 200, 'returns HTTP OK');
+  t.equal(response.result.includes('/path/param2'), true);
```

Comment:
Can we do a deepEqual instead?

Reply:
Will need to parse json, but will also make sure any extra routes don't get added